Cubical/Algebra/AbGroup/Properties.agda

```diff
@@ -24,3 +35,86 @@ module AbGroupTheory (A : AbGroup ℓ) where
+fst (addGroupHom A B ϕ ψ) x = AbGroupStr._+_ (snd B) (ϕ .fst x) (ψ .fst x)
+snd (addGroupHom A B ϕ ψ) = makeIsGroupHom
+  λ x y → cong₂ (AbGroupStr._+_ (snd B))
+                 (IsGroupHom.pres· (snd ϕ) x y)
+                 (IsGroupHom.pres· (snd ψ) x y)
+        ∙ AbGroupTheory.comm-4 B (fst ϕ x) (fst ϕ y) (fst ψ x) (fst ψ y)
+
+negGroupHom : (A B : AbGroup ℓ) (ϕ : AbGroupHom A B) → AbGroupHom A B
+fst (negGroupHom A B ϕ) x = AbGroupStr.-_ (snd B) (ϕ .fst x)
+snd (negGroupHom A B ϕ) =
+  makeIsGroupHom λ x y
+    → sym (IsGroupHom.presinv (snd ϕ) (AbGroupStr._+_ (snd A) x y))
+     ∙ cong (fst ϕ) (GroupTheory.invDistr (AbGroup→Group A) x y
+                  ∙ AbGroupStr.+Comm (snd A) _ _)
+     ∙ IsGroupHom.pres· (snd ϕ) _ _
+     ∙ cong₂ (AbGroupStr._+_ (snd B))
+             (IsGroupHom.presinv (snd ϕ) x)
+             (IsGroupHom.presinv (snd ϕ) y)
+
+subtrGroupHom : (A B : AbGroup ℓ) (ϕ ψ : AbGroupHom A B) → AbGroupHom A B
+subtrGroupHom A B ϕ ψ = addGroupHom A B ϕ (negGroupHom A B ψ)
+
+
+
+-- ℤ-multiplication defines a homomorphism for abelian groups
+private module _ (G : AbGroup ℓ) where
+  ℤ·isHom-pos : (n : ℕ) (x y : fst G)
+    → (pos n ℤ[ (AbGroup→Group G) ]· (AbGroupStr._+_ (snd G) x y))
+     ≡ AbGroupStr._+_ (snd G) ((pos n) ℤ[ (AbGroup→Group G) ]· x)
+                              ((pos n) ℤ[ (AbGroup→Group G) ]· y)
+  ℤ·isHom-pos zero x y =
+    sym (AbGroupStr.+IdR (snd G) (AbGroupStr.0g (snd G)))
+  ℤ·isHom-pos (suc n) x y =
+    cong₂ (AbGroupStr._+_ (snd G))
+          refl
+          (ℤ·isHom-pos n x y)
+    ∙ AbGroupTheory.comm-4 G _ _ _ _
+
+  ℤ·isHom-neg : (n : ℕ) (x y : fst G)
+    → (negsuc n ℤ[ (AbGroup→Group G) ]· (AbGroupStr._+_ (snd G) x y))
+     ≡ AbGroupStr._+_ (snd G) ((negsuc n) ℤ[ (AbGroup→Group G) ]· x)
+                              ((negsuc n) ℤ[ (AbGroup→Group G) ]· y)
+  ℤ·isHom-neg zero x y =
+    GroupTheory.invDistr (AbGroup→Group G) _ _
+    ∙ AbGroupStr.+Comm (snd G) _ _
+  ℤ·isHom-neg (suc n) x y =
+    cong₂ (AbGroupStr._+_ (snd G))
+          (GroupTheory.invDistr (AbGroup→Group G) _ _
+            ∙ AbGroupStr.+Comm (snd G) _ _)
+          (ℤ·isHom-neg n x y)
+    ∙ AbGroupTheory.comm-4 G _ _ _ _
+
+ℤ·isHom : (n : ℤ) (G : AbGroup ℓ) (x y : fst G)
+  → (n ℤ[ (AbGroup→Group G) ]· (AbGroupStr._+_ (snd G) x y))
+   ≡ AbGroupStr._+_ (snd G) (n ℤ[ (AbGroup→Group G) ]· x)
+                            (n ℤ[ (AbGroup→Group G) ]· y)
+ℤ·isHom (pos n) G = ℤ·isHom-pos G n
+ℤ·isHom (negsuc n) G = ℤ·isHom-neg G n
+
+-- left multiplication as a group homomorphism
+multₗHom : (G : AbGroup ℓ) (n : ℤ) → AbGroupHom G G
+fst (multₗHom G n) g = n ℤ[ (AbGroup→Group G) ]· g
+snd (multₗHom G n) = makeIsGroupHom (ℤ·isHom n G)
+
+-- Abelian groups quotiented by a natural number
+_/^_ : (G : AbGroup ℓ) (n : ℕ) → AbGroup ℓ
+G /^ n =
+  Group→AbGroup
+    ((AbGroup→Group G)
+     / (imSubgroup (multₗHom G (pos n))
+      , isNormalIm (multₗHom G (pos n)) (AbGroupStr.+Comm (snd G))))
```

Comment:
Hmm, this back and forth is a bit weird. Wouldn't it be nicer to have general lemmas for quotienting AbGroups? I have a vague recollection of wanting this when defining homology groups for chain complexes... Maybe I did it then?